plugins/snowflake/dbt/include/snowflake/macros/adapters.sql

```diff
@@ -1,14 +1,33 @@
 {% macro snowflake__create_table_as(temporary, relation, sql) -%}
   {%- set transient = config.get('transient', default=true) -%}
+  {# TODO: Add support for more than one keys #}
+  {%- set cluster_by_keys = config.get('cluster_by') -%}
+  {%- set is_incremental = config.get('materialized') == 'incremental' -%}
+  {%- if cluster_by_keys is not string and cluster_by_keys is not none -%}
+    {%-  set cluster_by_keys = cluster_by_keys|join(", ") -%}
+  {%- endif -%}
+
+      create or replace {% if temporary -%}
+        temporary
+      {%- elif transient -%}
+        transient
+      {%- endif %} table {{ relation }}
+      as (
+        {%- if cluster_by_keys is not none -%}
+          select * from(
+            {{ sql }}
+            ) order by ({{ cluster_by_keys }})
+        {%- else -%}
+          {{ sql }}
+        {%- endif %}
+      );
+    {% if cluster_by_keys is not none and is_incremental -%}
```

Comment:
I don't think we need this check on `is_incremental` here. Instead, I think it might make more sense to add a Snowflake-specific config called `automatic_clustering` which controls whether or not dbt enables automatic clustering. In particular, I think this will be useful in conjunction with dev/prod environments. It's probably not super desirable to always enable automatic clustering in dev (it can be expensive!). I think it's less important to check that the model is `incremental`, as automatic clustering simply won't run for non-incremental models (like `tables`) because dbt never runs DML against them!

You buy this?

I'm not even sure that we need to add this check -- if the user specifies a `cluster_by` for a non-incremental model (like a `table`), then there shouldn't be any harm in turning on automatic clustering.

I _am_ open to adding another config for Snowflake, like `automatic_clustering` which controls specifically when the `alter table {{relation}} resume recluster;` query is executed.

Reply:
I think I buy it! It makes sense that if there is no DML then no automatic clustering should be triggered, but I'm not sure if it wouldn't actually try to cluster things when you turn it on, which could be costly but I'm not sure. We'd have to have a snowflake specialist confirm this.

So, yes, I think putting `alter table {{relation}} resume recluster;` behind a config. 

Curious, how would you implement an override of automatic clustering if you were running your model against a `--target: dev` for example? You're hinting that that's configurable in the `dbt_project.yml`? or would you put an override in your `profile.yml`?

Reply:
(we discussed on Slack, but for posterity:)

Yeah  - that’s a separate issue that would be really good to address soon. When dbt evaluates the `models:` block in the `dbt_project.yml` file, it doesn’t include the specified `target` in the context. So, it would be great to do:
```
models:
   automatic_clustering: "{{ target.name == 'prod' }}"
```

but unfortunately, `target` is going to evaluate to `none` in this context currently, so that won’t work at the moment. Unsure if we have an issue for this at the moment, but I can create one